File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")

    n = number
    count = 0

    while n != 1:
        if n & 1:          # odd
            n = n * 3 + 1
        else:              # even
            n >>= 1        # divide by 2
        count += 1

    return count

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
	if n <= 0 {
		return 0, errors.New("n must be positive")
	}

	number := n
	steps := 0

	for number != 1 {
		if number%2 == 0 {
			number /= 2
		} else {
			number = number*3 + 1
		}

		steps++
	}

	return steps, nil
}